sklearn/metrics/_classification.py

```diff
@@ -382,6 +405,11 @@ def confusion_matrix(
             cm = cm / cm.sum()
         cm = np.nan_to_num(cm)
 
+    if pos_label is not None and pos_label != labels[-1]:
```

Comment:
what happens if labels is in reverse order? like `confusion_matrix(..., pos_label=0, labels=[1, 0])`

Reply:
I would expect `labels` to be used to select a subset of labels and `pos_label` to override the position if provided.

Reply:
If we are fine with that then, I can make a test. This would be important to do so.

Reply:
what would it mean to have a multiclass problem, select only two labels, and declare one as positive? Is that a sensible usecase? I think of labels as:

- making sure things work even if only one class is present in `y_true` / `y_pred`
- speed up things by not having to find labels from the data

Reply:
> what would it mean to have a multiclass problem, select only two labels, and declare one as positive? Is that a sensible usecase?

You will not be able to do that. `pos_label` will raise an error if `y_type` is multiclass before attempting any label selection.

I agree with you about the use-cases. I think that it comes in handy when dealing with strings due to potentially undesired alphabetic ordering:

```python
confusion_matrix(y_true, y_pred, labels=estimator.classes_, pos_label="positive_class")
```

`estimator.classes_` would be ordered alphabetically and it would be annoying to reorder.